pkg/model/resource/gvk.go

```diff
@@ -60,8 +53,9 @@ func (gvk GVK) Validate() error {
 	if gvk.Version == "" {
 		return fmt.Errorf(versionRequired)
 	}
-	if !versionRegex.MatchString(gvk.Version) {
-		return fmt.Errorf("Version must match %s (was %s)", versionPattern, gvk.Version)
+	if err := validation.IsDNS1035Label(gvk.Version); err != nil {
```

Comment:
Could you confirm if that is the right validation done by k8s?
@erikgb @varshaprasad96, could you lend a hand with this review?
That would be appreciate a lot

REALLY THANK YOU